### DISCUSSION TITLE
Is Notepad still a viable option for text and file editing in a world dominated by VSCode?

### DISCUSSION THREAD

**Discussion by Greatz08** [dail8859/NotepadNext#534] [General]
Notepad is pretty lightweight and good but in today's world where max use vscode for all text and other files editing,I know that program is electron based and little hard on ram and cpu usage sometime still its much much more capable compared to notepad so i want to know what all reason's are still there to use this project compared to vscode except its lightness in size. Plus can you tell what all changes or advancements are there in this project compared to normal notepad because i couldnt find that in main page and have to see many changelogs to understand differences so tell that too and mention those major changes in main page so people can find and understand that part easily instead of creating discussions :-)

**Comment by csells** [comment]
I don't know about you, but I have all kinds of text editing I do in a day that isn't code. If I've got VS Code running already, then a new window does the trick. But if I don't, I feel myself getting older waiting for VS Code to start. I love Notepad++ on Windows and would love to have it on my mac, too, just to get those precious moments back in my life.

**Comment by dail8859** [comment]
I really don't know how to answer this because it depends. You've already stated some obvious advantages and disadvantages.

It depends on the users, what they are comfortable with, use cases, what needs they have from a text editor and/or IDE, and numerous other reasons.

My suggestion is to try it and see how it works for your needs. If VSCode or Notepad++ works better then use those. I find myself between Notepad Next, VSCode, and Visual Studio on a daily basis.

I'm happy to answer more specific questions but just generally answering "what's different" is best left for you to find what is already there or missing.

**Comment by jonchines** [comment]
Personally, I don't use Notepad++, Notepadqq, or NotepadNext for code development (sometimes a quick edit) but I also don't use VSCode (its installed on my Windows and Linux machines but almost never used).  I do my development work (I am more on the Operations side of the house so Ansible, Terraform, Python, TypeScript, shell scripts, SQL) in Neovim.  I do, however, use Notepad++ in Windows for log viewing.  There is almost no other program that can reliably and quickly open and search large log files (350-1000MB).  Outside of tools like Klogg (which is a wildly different approach to log parsing), this has always been a chink in my Linux workflow.  I have tried numerous editors for this task - Kate, Geary, Vim, Neovim (treesitter disabled), Notepadqq, etc. - and none before NotepadNext have come close to Notepad++.  I am thankful, and so is my Linux workflow, that NotepadNext does this job so well.

That is a long way to say that I don't view Notepad???? as a development tool but it is certainly a valuable tool in my devops toolbox.

**Reply by dail8859** [reply]
> That is a long way to say that I don't view Notepad???? as a development tool but it is certainly a valuable tool in my devops toolbox.

Agreed. There is a big difference between a "source code editor" and a true "IDE". To be fair there is also alot of gray area and overlap in between those. So many people try to force Notepad++, Notepadqq, Notepad Next, etc into being an IDE, but they are not.

They are certainly useful, but they are not the right tool for every job.

**Reply by sadatned** [reply]
Notepad++ has long been a Swiss Army knife among text editors—nothing else truly comes close in terms of flexibility, features, and usability.

That’s why I’m genuinely thankful for the Notepad Next project, which brings a Notepad++-like experience to macOS. While it’s still missing a few key features like the Run and Plugins menus, it’s a big step forward for macOS users looking for a comparable tool.

👏 A big thanks to the developers of Notepad Next for making this possible!

**Comment by pcunite** [comment]
The super easy to use multi-tab search & replace is Notepad++ superpower. It's also just a really nice way to work with text files in general.
